index.js

```diff
@@ -1,175 +1,249 @@
-    '  if (typeof window === "undefined") { return };',
-    '  var id = "webpack-livereload-plugin-script-' + this.instanceId + '";',
-    '  if (document.getElementById(id)) { return; }',
-    '  var el = document.createElement("script");',
-    '  el.id = id;',
-    '  el.async = true;',
-    '  el.src = "' + this.protocol + '//' + this.hostname + ':' + this.port + '/livereload.js";',
-    '  document.getElementsByTagName("head")[0].appendChild(el);',
-    '}());',
-    ''
-  ].join('\n');
-};
-
-LiveReloadPlugin.prototype.scriptTag = function scriptTag(source) {
-  var js = this.autoloadJs();
-  if (this.options.appendScriptTag && this.isRunning) {
-    return js + source;
+
+  /**
+   * @returns {string}
+   * @private
+   */
+  _autoloadJs() {
+    return (
+        `
+        // webpack-livereload-plugin
+        (function() {
+          if (typeof window === "undefined") { return };
+          var id = "webpack-livereload-plugin-script-${this.instanceId}";
+          if (document.getElementById(id)) { return; }
+          var el = document.createElement("script");
+          el.id = id;
+          el.async = true;
+          el.src = "${this.options.protocol}//${this.options.hostname}:${this.options.port}/livereload.js";
+          document.getElementsByTagName("head")[0].appendChild(el);
+          console.log("[Live Reload] enabled");
+        }());
+        `
+    );
   }
-  else {
-    return source;
+
+  /**
+   * @param source
+   * @returns {*}
+   * @private
+   */
+  _scriptTag(source) {
+    if (this.options.appendScriptTag && this._isRunning()) {
+      return this._autoloadJs() + source;
+    }
+    else {
+      return source;
+    }
   }
-};
-
-LiveReloadPlugin.prototype.applyCompilation = function applyCompilation(compilation) {
-  compilation.mainTemplate.hooks.startup.tap('LiveReloadPlugin', this.scriptTag.bind(this));
-};
-
-LiveReloadPlugin.prototype.apply = function apply(compiler) {
-  this.compiler = compiler;
-  compiler.hooks.compilation.tap('LiveReloadPlugin', this.applyCompilation.bind(this));
-  compiler.hooks.watchRun.tapAsync('LiveReloadPlugin', this.start.bind(this));
-  compiler.hooks.done.tap('LiveReloadPlugin', this.done.bind(this));
-  compiler.hooks.failed.tap('LiveReloadPlugin', this.failed.bind(this));
-};
+
+  /**
+   * @param data
+   * @returns {boolean|*}
+   * @private
+   */
+  _fileIgnoredOrNotEmitted(data) {
+    const size = this.isWebpack4 ? data[1].emitted : data[1].size();
+
+    if (Array.isArray(this.options.ignore)) {
+      return !anymatch(this.options.ignore, data[0]) && size;
+    }
+
+    return !data[0].match(this.options.ignore) && size;
+  }
+
+  /**
+   * Check compiled source hash
+   *
+   * @param data
+   * @returns {boolean}
+   * @private
+   */
+  _fileHashDoesntMatch(data) {
+    if (!this.options.useSourceHash)
+      return true;
+
+    const sourceHash = LiveReloadPlugin.generateHashCode(data[1].source());
+
+    if (this.sourceHashs?.[data[0]] === sourceHash) {
```

Comment:
`this.sourceHashes` should always be defined so optional chaining shouldn't be needed. Removing also prevents a syntax error in node v12

```suggestion
    if (this.sourceHashs.[data[0]] === sourceHash) {
```